File: services/device/src/main/java/com/example/iter/device/dto/response/EquipmentListResponse.java
package com.example.iter.device.dto.response;

import java.util.List;

public record EquipmentListResponse(
        List<EquipmentSummaryResponse> content,
        int page,
        int size,
        long totalElements,
        int totalPages,
        boolean first,
        boolean last
) {
    public EquipmentListResponse {
        content = List.copyOf(content);
    }
}

File: services/device/src/main/kotlin/iter/device/dto/response/EquipmentListResponse.kt
package iter.device.dto.response

data class EquipmentListResponse(
    val content: List<EquipmentSummaryResponse>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
    val first: Boolean,
    val last: Boolean,
)
